src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -182,263 +182,211 @@ enum IntrinsicId
-				// CreateInstanceFrom (string assemblyFile, string typeName, bool ignoreCase, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder? binder, object? []? args, System.Globalization.CultureInfo? culture, object? []? activationAttributes)
-				// CreateInstanceFrom (string assemblyFile, string typeName, object? []? activationAttributes)
-				//
-				// CreateInstanceFromAndUnwrap (string assemblyFile, string typeName)
-				// CreateInstanceFromAndUnwrap (string assemblyFile, string typeName, bool ignoreCase, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder? binder, object? []? args, System.Globalization.CultureInfo? culture, object? []? activationAttributes)
-				// CreateInstanceFromAndUnwrap (string assemblyFile, string typeName, object? []? activationAttributes)
-				//
-				case var appDomainCreateInstanceMethodName when calledMethod.DeclaringType.Name == "AppDomain" && calledMethod.DeclaringType.Namespace == "System"
-					&& (appDomainCreateInstanceMethodName == "CreateInstance" || appDomainCreateInstanceMethodName == "CreateInstanceAndUnwrap"
-						|| appDomainCreateInstanceMethodName == "CreateInstanceFrom" || appDomainCreateInstanceMethodName == "CreateInstanceFromAndUnwrap")
-					&& calledMethod.Parameters.Count >= 2
-					&& calledMethod.Parameters [0].ParameterType.MetadataType == MetadataType.String
-					&& calledMethod.Parameters [1].ParameterType.MetadataType == MetadataType.String:
-					return appDomainCreateInstanceMethodName switch
-					{
-						"CreateInstance" => IntrinsicId.AppDomain_CreateInstance,
-						"CreateInstanceAndUnwrap" => IntrinsicId.AppDomain_CreateInstanceAndUnwrap,
-						"CreateInstanceFrom" => IntrinsicId.AppDomain_CreateInstanceFrom,
-						"CreateInstanceFromAndUnwrap" => IntrinsicId.AppDomain_CreateInstanceFromAndUnwrap,
-						_ => throw new ArgumentException ($"Reflection call '{calledMethod.FullName}' is unexpected.")
-					};
+			return method.DeclaringType.Name == typeName && method.DeclaringType.Namespace == namespaceName;
+		}
 
-				//
-				// System.Reflection.Assembly
-				//
-				// CreateInstance (string typeName)
-				// CreateInstance (string typeName, bool ignoreCase)
-				// CreateInstance (string typeName, bool ignoreCase, BindingFlags bindingAttr, Binder? binder, object []? args, CultureInfo? culture, object []? activationAttributes)
-				//
-				case "CreateInstance" when calledMethod.DeclaringType.Name == "Assembly" && calledMethod.DeclaringType.Namespace == "System.Reflection"
-					&& calledMethod.Parameters.Count >= 1
-					&& calledMethod.Parameters [0].ParameterType.MetadataType == MetadataType.String:
-					return IntrinsicId.Assembly_CreateInstance;
+		static bool HasParameterOfType (MethodDefinition method, int parameterIndex, string typeFullName)
+		{
+			return method.Parameters.Count > parameterIndex && method.Parameters [parameterIndex].ParameterType.FullName == typeFullName;
```

Comment:
Avoid using `FullName` it does string concatenation every time. I'd suggest using `IsTypeOf` as above